lib/tasks/helpers/newrelicyml.rb

```diff
@@ -0,0 +1,142 @@
+      # uncomment or edit them if they appear relevant to your application needs.
+
+  HEADER
+
+  FOOTER = <<~FOOTER
+    # Environment-specific settings are in this section.
+    # RAILS_ENV or RACK_ENV (as appropriate) is used to determine the environment.
+    # If your application has other named environments, configure them here.
+    development:
+      <<: *default_settings
+      app_name: <%= app_name %> (Development)
+
+    test:
+      <<: *default_settings
+      # It doesn't make sense to report to New Relic from automated test runs.
+      monitor_mode: false
+
+    staging:
+      <<: *default_settings
+      app_name: <%= app_name %> (Staging)
+
+    production:
+      <<: *default_settings
+  FOOTER
+
+  def self.get_configs(defaults)
+    final_configs = {}
+
+    defaults.sort.each do |key, value|
+      next if CRITICAL.include?(key) || SKIP.include?(key)
+
+      if public_config?(value)
+        sanitized_description = sanitize_description(value[:description])
+        description = format_description(sanitized_description)
+        default = default_value(key, value)
+        final_configs[key] = {description: description, default: default}
+      end
+    end
+
+    final_configs
+  end
+
+  def self.public_config?(value)
+    value[:public] == true
+  end
+
+  def self.sanitize_description(description)
+    # remove callouts
+    description = description.split("\n").reject { |line| line.match?('</?Callout') }.join("\n")
+    # remove InlinePopover, keep the text inside type
+    description.gsub!(/<InlinePopover type="(.*)" \/>/, '\1')
+    # remove hyperlinks
+    description.gsub!(/\[([^\]]+)\]\([^\)]+\)/, '\1')
+    # remove single pairs of backticks
+    description.gsub!(/`([^`]+)`/, '\1')
+    # removed hrefs links
+    description.gsub!(/<a href="(.*)">(.*)<\/a>/, '\2')
+
+    description
+  end
+
+  def self.format_description(description)
+    # remove leading and trailing whitespace
+    description = description.strip
+    # wrap text after 80 characters
+    description.gsub!(/(.{1,80})(\s+|\Z)/, "\\1\n")
+    # add hashtags to lines
+    description = description.split("\n").map { |line| "  # #{line}" }.join("\n")
+
+    description
+  end
+
+  def self.default_value(key, config_hash)
+    if PROCS.include?(key)
+      PROCS[key]
+    else
+      default = config_hash[:documentation_default].nil? ? config_hash[:default] : config_hash[:documentation_default]
+      default = 'nil' if default.nil?
+      default = '""' if default == ''
+
+      default
+    end
+  end
+
+  def self.build_string(defaults)
+    configs = get_configs(defaults)
+    yml_string = ''
+
+    configs.each do |key, value|
+      yml_string += "#{value[:description]}\n  # #{key}: #{value[:default]}\n\n"
+    end
+
+    yml_string
+  end
+
+  def self.write_file(defaults = DEFAULTS)
+    File.write('newrelic.yml', HEADER + build_string(defaults) + FOOTER)
```

Comment:
It would be helpful if you could include as a PR comment a diff of how a newly generated .yml file looks compared to what we have now.

Reply:
This LOC is the only one that looks like it isn't covered by tests according to Simplecov.

Reply:
In my rake task PR, I also didn't test the lines that did the actual file writing and reading, instead I just tested that what I was planning to write was correct.

Reply:
For brand new code...

ideally we would either a) have 100% code coverage, or b) use `:nocov:` sparingly to make it explicitly clear that we don't want to bother with covering something. See [skipping code](https://github.com/simplecov-ruby/simplecov#ignoringskipping-code).

Reply:
Are we okay with skipping code coverage on rake tasks? Or perhaps we can pair on it.

Reply:
I think this is okay since everything else is so well-tested. 

I'm also fine with James's recommendation to use `:nocov:` where we're aware test coverage is missing and intentionally didn't add it.